src/V3EmitC.cpp

```diff
@@ -1012,6 +1012,18 @@ class EmitCStmts : public EmitCBaseVisitor {
     }
     virtual void visit(AstStreamL* nodep) override {
         // Attempt to use a "fast" stream function for slice size = power of 2
+        if (VN_IS(nodep->lhsp()->dtypep(), UnpackArrayDType)) {
+            const AstUnpackArrayDType* arrayp= VN_CAST_CONST(nodep->lhsp()->dtypep(), UnpackArrayDType);
+            string prefix = "VL_STREAML_";
+            switch(arrayp->subDTypep()->width()) {
+            case 8:  prefix = "VL_STREAML_UA_BYTE";break;
+            case 16: prefix = "VL_STREAML_UA_SHORTINT"; break;      
+            case 32: prefix = "VL_STREAML_UA_INT";break;
+            case 64: prefix = "VL_STREAML_UA_LONGINT"; break;  
```

Comment:
What if size is none of these?  Is that possible?  If not please assert.

Verilator internally generally doesn't use BYTE/SHORTING/INT/LONGINT as names in emitted code, it uses the letters C/S/I/Q as a C data may represent e.g. 7 bits not just 8.

typedef vluint8_t    CData;     ///< Verilated pack data, 1-8 bits
typedef vluint16_t   SData;     ///< Verilated pack data, 9-16 bits
typedef vluint32_t   IData;     ///< Verilated pack data, 17-32 bits
typedef vluint64_t   QData;     ///< Verilated pack data, 33-64 bits

Reply:
please provide a name example. Right now they are:
UA - stands for unpacked array 
III - is generated

VL_STREAML_UA_BYTE_III
VL_STREAML_UA_SHORTINT_QQI   
VL_STREAML_UA_INT_WWI   
VL_STREAML_UA_LONGINT_WWI

Reply:
The letters are the argument types, so for

`VL_STREAML_UA_BYTE_III(int a, int lbits, int b, unsigned char ld[], IData rd) VL_PURE {`

this should be prototyped

`VL_STREAML_UA_BYTE_CI(int, int lbits, int, CData ld[], IData rd) VL_PURE {`

as it takes two args, char data (C per above) and int data (I per above).

Reply:
ok , so I can keep VL_STREAML_UA_BYTE prefix

Reply:
Sorry, I misdid that,  There's no reason for BYTE as is the "C".  Edited above.

Reply:
Thanks

Reply:
done